newscoop/src/Newscoop/NewscoopBundle/Security/Http/Authentication/AuthenticationSuccessHandler.php

```diff
@@ -44,9 +46,10 @@ function onAuthenticationSuccess(Request $request, TokenInterface $token)
     {
         $user = $token->getUser();
 
+        // This should actually be handle by the AuthenticationFailedHandler
         if (!$user->isAdmin()) { // can't go into admin
-            $redirector = \Zend_Controller_Action_HelperBroker::getStaticHelper('redirector');
-            $redirector->direct('index', 'index', 'default');
+            $request->getSession()->set(SecurityContextInterface::AUTHENTICATION_ERROR, new AuthenticationException('User is not an admin.'));
+            return $this->httpUtils->createRedirectResponse($request, 'admin_login');
```

Comment:
:) i couldnt find a way to let user who are not admin go through the AuthenticationFailureHandler, since Symfony doesn't provide basic methods for that. Maybe if we write a custom UserProvider it could be possible, but it also works this way.